pkg/cmd/workflow/shared/shared_test.go

```diff
@@ -0,0 +1,422 @@
+					httpmock.REST("GET", "repos/OWNER/REPO/actions/workflows"),
+					httpmock.StatusJSONResponse(200, WorkflowsPayload{
+						Workflows: []Workflow{
+							{
+								ID:    1,
+								Name:  "Workflow Name",
+								State: Active,
+							},
+						}}),
+				)
+			},
+			expectedWorkflow: Workflow{
+				ID:    1,
+				Name:  "Workflow Name",
+				State: "active",
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			reg := &httpmock.Registry{}
+			if tt.httpStubs != nil {
+				tt.httpStubs(reg)
+			}
+			client := api.NewClientFromHTTP(&http.Client{Transport: reg})
+
+			workflow, err := FindWorkflow(client, tt.repo, tt.workflowSelector, tt.states)
+			if tt.expectedError != nil {
+				require.Error(t, err)
+				assert.Equal(t, tt.expectedError, err)
+			} else if tt.expectedHTTPError != nil {
+				var httpErr api.HTTPError
+				require.ErrorAs(t, err, &httpErr)
+				assert.Equal(t, tt.expectedHTTPError.Error(), httpErr.Error())
+			} else {
+				require.NoError(t, err)
+				assert.Equal(t, tt.expectedWorkflow, workflow[0])
+			}
+		})
+	}
+}
+
+type ErrorTransport struct {
+	Err error
+}
+
+func (t *ErrorTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	return nil, t.Err
+}
+
+func TestFindWorkflow_nonHTTPError(t *testing.T) {
+	t.Run("When the client fails to instantiate, it returns the error", func(t *testing.T) {
+		client := api.NewClientFromHTTP(&http.Client{Transport: &ErrorTransport{Err: errors.New("non-HTTP error")}})
+		repo := ghrepo.New("OWNER", "REPO")
+		workflow, err := FindWorkflow(client, repo, "1", nil)
+
+		require.Error(t, err)
+		assert.ErrorContains(t, err, "non-HTTP error")
+		assert.Nil(t, workflow)
+	})
+}
+
+func Test_getWorkflowsByName_filtering(t *testing.T) {
+	tests := []struct {
+		name              string
+		workflowName      string
+		repo              ghrepo.Interface
+		states            []WorkflowState
+		httpStubs         func(*httpmock.Registry)
+		expectedWorkflows []Workflow
+		expectedErrorMsg  string
+	}{
+		{
+			name:         "When no workflows match, no workflows are returned",
+			workflowName: "Unmatched Workflow Name",
+			repo:         ghrepo.New("OWNER", "REPO"),
+			states:       []WorkflowState{Active},
+			httpStubs: func(reg *httpmock.Registry) {
+				reg.Register(
+					httpmock.REST("GET", "repos/OWNER/REPO/actions/workflows"),
+					httpmock.StatusJSONResponse(200, WorkflowsPayload{
+						Workflows: []Workflow{
+							{
+								ID:    1,
+								Name:  "Workflow Name",
+								State: Active,
+							},
+							{
+								ID:    2,
+								Name:  "Workflow Name",
+								State: DisabledInactivity,
+							},
+							{
+								ID:    3,
+								Name:  "Workflow Name",
+								State: Active,
+							},
+						},
+					}),
+				)
+			},
+			expectedWorkflows: []Workflow(nil),
+		},
+		{
+			name:         "When there are more than one workflow with the same name, only the ones matching the provided state are returned",
```

Comment:
🤔 Previously, I haven't considered whether Actions will allow you to have the same workflow with multiple names and states, however I just confirmed it does.

![neat-bender](https://github.com/user-attachments/assets/e784db2c-7c03-42ea-9e27-bd6fbe1346c5)